Lib/collections/__init__.py

```diff
@@ -979,6 +979,28 @@ def __contains__(self, key):
 
     # Now, add the methods in dicts but not in MutableMapping
     def __repr__(self): return repr(self.data)
+
+    def __or__(self, other):
+        if isinstance(other, UserDict):
+            return self.__class__(self.data | other.data)
+        if isinstance(other, type(self.data)):
+            return self.__class__(self.data | other)
+        return self.__class__(self.data | dict(other))
```

Comment:
I'm surprised there's no branch that returns `NotImplemented`. What if there was some other class that knew how to 'or' itself with `UserDict`?

Reply:
In all honesty, I was just trying to shut up the failing "make sure we have all of the `dict` methods" test.  I've never worked with these types before, so I copied-and-pasted the logic from [the `__add__` trio in `UserList`](https://github.com/python/cpython/blob/master/Lib/collections/__init__.py#L1044-L1063)

Reply:
That's ancient code, probably older than `NotImplemented`. And nothing can be done about it without breaking somebody's working code. But here we're adding fresh code, so let's do this right.